@xen-orchestra/rest-api/src/middlewares/generic-error-handler.middleware.mts

```diff
@@ -0,0 +1,38 @@
+import { createLogger } from '@xen-orchestra/log'
+import {
+  featureUnauthorized,
+  forbiddenOperation,
+  invalidParameters,
+  noSuchObject,
+  unauthorized,
+} from 'xo-common/api-errors.js'
+import { NextFunction, Request, Response } from 'express'
+
+const log = createLogger('xo:rest-api:error-handler')
+
+// must have 4 parameters to be recognized as an error middleware by express
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export default function genericErrorHandler(error: unknown, req: Request, res: Response, _next: NextFunction) {
+  if (!(error instanceof Error)) {
+    log.error(error)
+    res.status(500).json({ error })
+    return
+  }
+
+  if (noSuchObject.is(error)) {
+    res.status(404)
+  } else if (forbiddenOperation.is(error) || featureUnauthorized.is(error)) {
+    res.status(403)
+  } else if (unauthorized.is(error)) {
+    res.status(401)
+  } else if (invalidParameters.is(error)) {
+    res.status(400)
```

Comment:
I think this is a 422
https://http.cat/status/422
> 
   The 422 (Unprocessable Entity) status code means the server
   understands the content type of the request entity (hence a
   415(Unsupported Media Type) status code is inappropriate), and the
   syntax of the request entity is correct (thus a 400 (Bad Request)
   status code is inappropriate) but was unable to process the contained
   instructions.  For example, this error condition may occur if an XML
   request body contains well-formed (i.e., syntactically correct), but
   semantically erroneous, XML instruction
   
   https://www.rfc-editor.org/rfc/rfc4918#section-11.2